Now returning an error package when receiving the wrong block number

1c3t3a opened the pull request:
Closes #35. The connection::put Method now sends an error package before ending the communication in case the current_block count is not plausible.

1c3t3a commented:
I dont't really get how he's not finding [std::io::Error](https://doc.rust-lang.org/std/io/struct.Error.html) in the [error log](https://travis-ci.com/github/connorkuehl/tftp/jobs/395267498).

connorkuehl commented:
@1c3t3a it's because your development branch is out-of-date with my branch and that import statement might have disappeared since then. No worries! This is easily fixed by rebasing on top of my branch.

When you're on your local branch:

```console
$ git fetch https://github.com/connorkuehl/tftp main
$ git rebase FETCH_HEAD
```

There might be some merge conflicts that need to be resolved.

1c3t3a commented:
> @1c3t3a it's because your development branch is out-of-date with my branch and that import statement might have disappeared since then. No worries! This is easily fixed by rebasing on top of my branch.
> 
> When you're on your local branch:
> 
> ```
> $ git fetch https://github.com/connorkuehl/tftp main
> $ git rebase FETCH_HEAD
> ```
> 
> There might be some merge conflicts that need to be resolved.

Ah I see! I rebased and added the import again, now it should work fine!

connorkuehl commented on src/connection.rs:
```diff
@@ -69,7 +69,14 @@ impl Connection {
 
-            assert_eq!(Block::new(current_block), ack.body.block);
+            if Block::new(current_block) != ack.body.block {
+                let error = Packet::<Error>::error(
+                    Code::IllegalOperation,
+                    "Got an ack with the wrong packet number, terminate excecution",
+                );
```
```suggestion
                let error = Packet::error(
                    Code::IllegalOperation,
                    format!("expected ACK for {} but got ACK for {}", current_block, ack.body.block),
                );
```

Awesome! Just a couple of tweaks. I don't think we have to turbofish `Packet::<Error>::error`, I think Rust knows which `Packet<T>` we mean when we use `Packet::error`.

I tweaked the error message to include more info.

I haven't tried compiling these suggestions so maybe a bit of massaging is necessary.

connorkuehl reviewed (approved):
Thank you so much! :slightly_smiling_face:
